packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -54,70 +51,60 @@ export async function executePlugin(
   observer?: ProcessObserver,
 ): Promise<PluginReport> {
   const {
-    slug,
-    title,
-    icon,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    runner: onlyUsedForRestingPluginMeta,
+    audits: pluginConfigAudits,
     description,
     docsUrl,
-    version,
-    packageName,
     groups,
+    ...pluginMeta
   } = pluginConfig;
   const { args, command } = pluginConfig.runner;
 
-  const { duration, date } = await executeProcess({
+  const { date, duration } = await executeProcess({
     command,
     args,
     observer,
   });
+  const executionMeta = { date, duration };
 
-  try {
-    const processOutputPath = join(
-      process.cwd(),
-      pluginConfig.runner.outputFile,
-    );
+  const processOutputPath = join(process.cwd(), pluginConfig.runner.outputFile);
+
+  // read process output from file system and parse it
+  let unknownAuditOutputs = await readJsonFile<Record<string, unknown>[]>(
+    processOutputPath,
+  );
 
-    // read process output from file system and parse it
-    const auditOutputs = auditOutputsSchema.parse(
-      JSON.parse((await readFile(processOutputPath)).toString()),
+  // parse transform unknownAuditOutputs to auditOutputs
+  if (pluginConfig.runner?.outputTransform) {
+    unknownAuditOutputs = await pluginConfig.runner.outputTransform(
+      unknownAuditOutputs,
     );
+  }
+
+  // validate audit outputs
+  const auditOutputs = auditOutputsSchema.parse(unknownAuditOutputs);
+
+  // validate auditOutputs
+  auditOutputsCorrelateWithPluginOutput(auditOutputs, pluginConfigAudits);
 
-    const audits = auditOutputs.map(auditOutput => {
-      const auditMetadata = pluginConfig.audits.find(
-        audit => audit.slug === auditOutput.slug,
-      );
-      if (!auditMetadata) {
-        throw new PluginOutputError(
-          slug,
-          new Error(
-            `Audit metadata not found for slug ${auditOutput.slug} from runner output`,
-          ),
-        );
-      }
-      return {
+  // enrich `AuditOutputs` to `AuditReport`
+  const audits: AuditReport[] = auditOutputs.map(
+    (auditOutput: AuditOutput) =>
+      ({
         ...auditOutput,
-        ...auditMetadata,
-      };
-    });
-
-    // @TODO consider just resting/spreading the values
-    return {
-      version,
-      packageName,
-      slug,
-      title,
-      icon,
-      date,
-      duration,
-      audits,
-      ...(description && { description }),
-      ...(docsUrl && { docsUrl }),
-      ...(groups && { groups }),
-    } satisfies PluginReport;
-  } catch (error) {
-    const e = error as Error;
-    throw new PluginOutputError(slug, e);
-  }
+        ...pluginConfigAudits.find(audit => audit.slug === auditOutput.slug),
+      } as unknown as AuditReport),
```

Comment:
The `as unknown as` isn't very safe to use. In this case there's a more precise way by just casting the result of `find` (since you know you checked it already with `auditOutputsCorrelateWithPluginOutput`). That way it will still report other causes of errors.

```suggestion
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        ...pluginConfigAudits.find(audit => audit.slug === auditOutput.slug)!,
      }),
```

Reply:
fixed it differently